### PR TITLE
Adjust search_and_parse_results and search_judgments_and_parse_result…

### PR DESCRIPTION
**Pull request by anthonyhashemi**
Whilst working on https://github.com/nationalarchives/ds-caselaw-custom-api-client/pull/216, realised that it would make more sense to keep parsing of response objects away from the Client class.

Therefore:
## Changes in this PR:
- Adjusted search_and_parse_results and search_judgments_and_parse_results s to be search_and_decode_response and search_judgments_and_decode_response respectively

## Trello card / Rollbar error (etc)

https://trello.com/c/T6x6GXmc/803-ensure-judgments-dont-display-press-summaries-erroneously

### CODE REVIEW

File: tests/client/test_search_results.py [hunk 1]
```diff
@@ -1,54 +0,0 @@
-import pytest
-from lxml import etree
-
-from caselawclient.models.search_results import SearchResults
-
-
-class TestSearchResults:
```
Comment (anthonyhashemi) [inline tests/client/test_search_results.py:7]: These get added back in https://github.com/nationalarchives/ds-caselaw-custom-api-client/pull/223/files#diff-963e91dbe51e70ef86a30719fdb422d993fdb30e1a553bb10e444222c1f072c7

File: src/caselawclient/models/search_results.py [hunk 2]
```diff
@@ -1,52 +0,0 @@
-from typing import List
-
-from lxml import etree
-
-
-class SearchResults:
```
Comment (anthonyhashemi) [inline src/caselawclient/models/search_results.py:6]: This gets added back in https://github.com/nationalarchives/ds-caselaw-custom-api-client/pull/223/files#diff-05cdf70bd9924bdbcf34f282af03471503bbfc4db7634f87d89ce246acabda62

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,8 +7,7 @@ The format is based on [Keep a Changelog 1.0.0].
 ## [Unreleased]
 - Added `SearchParameters` dataclass for use with search functions using the legacy kwargs from `Client.advanced_search` and new `collections` field for filtering by collections
 - Changed `Client.advanced_search` interface to take in `SearchParameters` as opposed to the legacy kwargs.
-- Added `search_and_parse_results` and `search_judgments_and_parse_results` methods to `Client`
-- Added `SearchResults` class to encapsulate document search results
```
Comment (anthonyhashemi) [inline CHANGELOG.md:11]: This gets added back in https://github.com/nationalarchives/ds-caselaw-custom-api-client/pull/223/files

File: src/caselawclient/Client.py [hunk 4]
```diff
@@ -706,18 +705,15 @@ def get_properties_for_search_results(self, judgment_uris: list[str]) -> str:
...
-    def search_and_parse_results(
-        self, search_parameters: SearchParameters
-    ) -> SearchResults:
+    def search_and_decode_response(self, search_parameters: SearchParameters) -> str:
         response = self.advanced_search(search_parameters)
-        response_string = decode_multipart(response)
-        return SearchResults(response_string)
+        return decode_multipart(response)
 
-    def search_judgments_and_parse_results(
```
Comment (dragon-dxw) [inline src/caselawclient/Client.py:716]: Wary that the old names might be used somewhere, but I can't find any references to `search_and_parse_results` or `search_judgments_and_parse_results` outside the custom-api-client, so this looks OK.

**Review by dragon-dxw — APPROVED** [review]
Looks good, except we might not be triggering the test.

File: tests/client/test_search_and_decode_response.py [hunk 5]
```diff
@@ -1,27 +1,30 @@
...
 
 from caselawclient.Client import api_client
-from caselawclient.models.search_results import SearchResults
 from caselawclient.search_parameters import SearchParameters
 
 
 @patch("caselawclient.Client.api_client.advanced_search")
-def test_search_judgments_and_parse_results(
-    mock_advanced_search, mock_search_results_response
+def search_judgments_and_decode_response(
```
[the remark this answers is not in the document]
Comment (anthonyhashemi) [reply tests/client/test_search_and_decode_response.py:8]: Ah good catch, it must have happened when I was updating the names of the main functions.

updating now, cheers

File: src/caselawclient/Client.py (hunk 4 above)
Comment (anthonyhashemi) [reply src/caselawclient/Client.py:716]: I introduced them in previous PR, the client hasnt beeen released yet with these new functions

**Pull request merged**